staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -17,286 +17,44 @@ limitations under the License.
-func (s *SpdyRoundTripper) dialWithSocks5Proxy(req *http.Request, proxyURL *url.URL) (net.Conn, error) {
-	// ensure we use a canonical host with proxyReq
-	targetHost := netutil.CanonicalAddr(req.URL)
-	proxyDialAddr := netutil.CanonicalAddr(proxyURL)
-
-	var auth *proxy.Auth
-	if proxyURL.User != nil {
-		pass, _ := proxyURL.User.Password()
-		auth = &proxy.Auth{
-			User:     proxyURL.User.Username(),
-			Password: pass,
-		}
-	}
-
-	dialer := s.Dialer
-	if dialer == nil {
-		dialer = &net.Dialer{
-			Timeout: 30 * time.Second,
-		}
-	}
-
-	proxyDialer, err := proxy.SOCKS5("tcp", proxyDialAddr, auth, dialer)
-	if err != nil {
-		return nil, err
-	}
-
-	// According to the implementation of proxy.SOCKS5, the type assertion will always succeed
-	contextDialer, ok := proxyDialer.(proxy.ContextDialer)
-	if !ok {
-		return nil, errors.New("SOCKS5 Dialer must implement ContextDialer")
-	}
-
-	proxyDialConn, err := contextDialer.DialContext(req.Context(), "tcp", targetHost)
-	if err != nil {
-		return nil, err
-	}
-
-	if req.URL.Scheme == "https" {
-		return s.tlsConn(req.Context(), proxyDialConn, targetHost)
-	}
-	return proxyDialConn, nil
-}
-
-// tlsConn returns a TLS client side connection using rwc as the underlying transport.
-func (s *SpdyRoundTripper) tlsConn(ctx context.Context, rwc net.Conn, targetHost string) (net.Conn, error) {
-
-	host, _, err := net.SplitHostPort(targetHost)
-	if err != nil {
-		return nil, err
-	}
-
-	tlsConfig := s.tlsConfig
-	switch {
-	case tlsConfig == nil:
-		tlsConfig = &tls.Config{ServerName: host}
-	case len(tlsConfig.ServerName) == 0:
-		tlsConfig = tlsConfig.Clone()
-		tlsConfig.ServerName = host
-	}
-
-	tlsConn := tls.Client(rwc, tlsConfig)
-
-	if err := tlsConn.HandshakeContext(ctx); err != nil {
-		tlsConn.Close()
-		return nil, err
-	}
-
-	return tlsConn, nil
-}
-
-// dialWithoutProxy dials the host specified by url, using TLS if appropriate.
-func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (net.Conn, error) {
-	dialAddr := netutil.CanonicalAddr(url)
-	dialer := s.Dialer
-	if dialer == nil {
-		dialer = &net.Dialer{}
-	}
-
-	if url.Scheme == "http" {
-		return dialer.DialContext(ctx, "tcp", dialAddr)
+		rt: rt,
 	}
-
-	tlsDialer := tls.Dialer{
-		NetDialer: dialer,
-		Config:    s.tlsConfig,
-	}
-	return tlsDialer.DialContext(ctx, "tcp", dialAddr)
 }
 
-// proxyAuth returns, for a given proxy URL, the value to be used for the Proxy-Authorization header
-func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {
-	if proxyURL == nil || proxyURL.User == nil {
-		return ""
-	}
-	credentials := proxyURL.User.String()
-	encodedAuth := base64.StdEncoding.EncodeToString([]byte(credentials))
-	return fmt.Sprintf("Basic %s", encodedAuth)
+func (s *SpdyRoundTripper) WrappedRoundTripper() http.RoundTripper {
+	return s.rt
 }
 
 // RoundTrip executes the Request and upgrades it. After a successful upgrade,
```

Comment:
This comment looks wrong, is it talking about `NewConnection`?

Reply:
Fixed.